ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java

```diff
@@ -388,12 +388,18 @@ public static void process_attestations(BeaconState state, List<Attestation> att
               "process_attestations: Attestation source error 2");
           state.getPrevious_epoch_attestations().add(pendingAttestation);
         }
-
-        // Check signature
-        checkArgument(
-            is_valid_indexed_attestation(state, get_indexed_attestation(state, attestation)),
-            "process_attestations: Check signature");
       }
+
+      attestations.stream()
+          .parallel()
+          .forEach(
+              attestation -> {
+                // Check signature
+                checkArgument(
+                    is_valid_indexed_attestation(
+                        state, get_indexed_attestation(state, attestation)),
+                    "process_attestations: Check signature");
```

Comment:
Not critical, but would we be better off using `allMatch(is_valid_indexed_attestation(...))` here? I suspect the stream bails out if one thread throws an exception anyway but `allMatch` makes it a bit more explicit and then we can just check the result instead of catching the exception.

Could even use `.filter(!is_valid_indexed_attestation).findAny()` and include actual information about which attestation we found was invalid...

Reply:
@ajsutton yep, your variant looks better. 
What's the better way to address this change: reopen this PR, create a separate new one, or just put this change together with one of next PRs?

Reply:
Probably just a new PR unless you happen to be making more changes right in this area.

Reply:
#1115